### PR TITLE
Merge Initial Datasets Service Implementation 

### PR DESCRIPTION
**Pull request by dashaylan**
This PR is a nearly complete implementation of the datasets service. The code follows the Python Model Service and rnaget implementations.

Routes not implemented:
* /datasets/discover/search
* /datasets/discover/search/filter

Since access lists aren't implemented yet for the microservices, the search and discover/search would be doing the same thing.

The OpenAPI spec has been altered to put /changelog/{version} and /getVersions behind /datasets due to federation needing a uniform access point for each individual microservice.

*  /changelog/{version} --> /datasets/changelog/{version}
* /getVersions --> /datasets/getVersions

### CODE REVIEW

**Review by jimmyhli — APPROVED** [review]
Thanks for putting this together!

I wrote the specs here - theres a few things I'm not sure  will exist, but changing the following would involve the change of the specs
- A /DELETE /dataset endpoint. The related db operation method will certainly still be used, but it'd be a matter of the operation being invoked from an endpoint or a cli command
- How the dataset_id is generated, and the uuid of the dataset across sites (was discussed yesterday), it would affect how we do federation here 
- We need to brainstorm on the best way to incorporate DUO here into the spec/model i guess

File: datasets.ini [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[uwsgi]
+module = wsgi:application
+chdir = /home/dnaidoo/Documents/datasets_service
```
Comment (zbozoky) [inline datasets.ini]: this path might need a change to make it universal

**Review by zbozoky — APPROVED** [review]
Great work Dash, thank you!

File: datasets.ini (hunk 1 above)
Comment (dashaylan) [reply datasets.ini]: Turns out uwsgi has some magic variables that can be used. Changed both the chdir and socket paths to use a PWD command.

**Pull request merged**